File: ecopoints/app/src/main/java/at/htl/ecopoints/RankingActivity.java
package at.htl.ecopoints;

import android.os.Bundle;
import android.util.Log;

import androidx.activity.ComponentActivity;

import javax.inject.Inject;

import at.htl.ecopoints.ui.layout.RankingView;
import dagger.hilt.android.AndroidEntryPoint;

@AndroidEntryPoint
public class RankingActivity extends ComponentActivity {

    private static final String TAG = RankingActivity.class.getSimpleName();

    @Inject
    RankingView rankingView;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        Log.i(TAG, "onCreate()");
        super.onCreate(savedInstanceState);

        rankingView.compose(this);
    }
}

File: ecopoints/app/src/main/java/at/htl/ecopoints/activity/ui/theme/RankingActivity.kt
package at.htl.ecopoints.activity.ui.theme

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.ui.Modifier
import androidx.compose.ui.tooling.preview.Preview
import at.htl.ecopoints.activity.ui.theme.ui.theme.EcoPointsTheme
import at.htl.ecopoints.navigation.BottomNavBar

class RankingActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            val (currentScreen, setCurrentScreen) = remember { mutableStateOf("Ranking") }
            at.htl.ecopoints.activity.ui.theme.EcoPointsTheme {
                // A surface container using the 'background' color from the theme
                Surface(
                    modifier = Modifier.fillMaxSize(),
                    color = MaterialTheme.colorScheme.background
                ) {
                    Greeting2("Ranking")
                    Box {

                        BottomNavBar(
                            currentScreen = currentScreen,
                            onScreenSelected = { newScreen -> setCurrentScreen(newScreen) },
                            context = this@RankingActivity
                        )
                    }
                }
            }
        }
    }
}

@Composable
fun Greeting2(name: String, modifier: Modifier = Modifier) {
    Text(
        text = "Hello $name!",
        modifier = modifier
    )
}

@Preview(showBackground = true)
@Composable
fun GreetingPreview2() {
    EcoPointsTheme {
        Greeting2("Ranking")
    }
}